docs/contrib/PlatformBringUp.md

```diff
@@ -0,0 +1,415 @@
+
+For Example:
+
+
+```
+    DESIGN_NAME
+    PLATFORM
+    VERILOG_FILES
+    SDC_FILE
+    CORE_UTILIZATION
+    CORE_ASPECT_RATIO
+    CORE_MARGIN
+    PLACE_DENSITY
+```
+
+Alternatively, in place of `CORE_UTILIZATION` and `CORE_ASPECT_RATIO`, `DIE_AREA` and `CORE_AREA` can be specified instead.
+
+
+Following is a sample `config.mk` file for the `riscv32i` design:
+
+
+```
+#config.mk
+###########################
+export DESIGN_NAME     = riscv
+export PLATFORM        = sky130hd
+
+export VERILOG_FILES = $(sort $(wildcard ./designs/src/$(DESIGN_NICKNAME)/*.v))
+
+export SDC_FILE = ./designs/$(PLATFORM)/$(DESIGN_NICKNAME)/constraint.sdc
+
+export CORE_UTILIZATION  = 30
+export CORE_ASPECT_RATIO = 1
+export CORE_MARGIN       = 2
+export PLACE_DENSITY     = 0.70
+```
+
+
+Refer to the [OpenSTA][https://github.com/The-OpenROAD-Project/OpenSTA/blob/master/doc/OpenSTA.pdf] documenation to change these variables to suit the needs of the specific design being made.
+
+
+## constraint.sdc
+
+The `constraint.sdc` file defines timing constraints for the design. Here’s an example of a constraint.sdc file which defines a clock `clk` with a period of 8.4 nanoseconds.
+
+
+```
+#constraint.sdc
+############################
+create_clock [get_ports clk] -period 8.4
+```
+
+
+The `create_clock` command allows you to define clocks that are either connected to nets  or are virtual and can be customized.
+
+
+```
+#Create a netlist clock
+create_clock -period <float> <netlist clock list of regexes>
+
+#Create a virtual clock
+create_clock -period <float> -name <virtual clock name>
+
+#Create a netlist clock with custom waveform/duty-cycle
+create_clock -period <float> -waveform {rising_edge falling_edge} <netlist clock list of regexes>
+```
+
+
+
+## Liberty, LEF, and GDS Files
+
+The liberty, LEF, and GDS files do not technically have to reside inside the platform directory of respective technology as long as the paths set in the `config.mk` file point to the correct files. However, it is good practice to have all relevant files in one localized directory. The `.lib`, `.lef`, and `.gds` reside in directories named respectively for the specific technology.
+
+For example:
+
+
+```
+$ mdkir OpenROAD-flow-scripts/flow/platforms/MyNewPlatform/lib
+$ mdkir OpenROAD-flow-scripts/flow/platforms/MyNewPlatform/lef
+$ mdkir OpenROAD-flow-scripts/flow/platforms/MyNewPlatform/gds
+```
+
+
+A merged GDS file may be used instead of adding every individual `.gds` file from the standard cell library.
+
+Once the liberty file, tech and macro LEF files, and either the merged standard cell GDS or individual standard cell GDS files have been generated, place them in their respective directories and set the `lib`, `lef`, and `gds` variables in the platform `config.mk` file to the correct paths.
+
+
+## Behavioral Models
+
+OpenROAD requires behavioral Verilog modules for a latch and a gated clock. These modules are used during synthesis and follow a specific naming convention.
```

Comment:
OpenROAD Flow. It is Yosys that requires it, not OpenROAD.

Reply:
Changed to Yosys